### PR TITLE
Add proxy option, example configuration.

### PR DESCRIPTION
**Pull request by ytjohn**
As you have the most complete fork, I wanted to merge my proxy additions in with yours.

This is my first time working with golang, so more than happy to improve.

### CODE REVIEW

File: slackcat-example.conf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "proxy": "http://proxy.example.com:3128",
+  "username": "slackcat",
+  "channel": "#slackcat",
+  "icon_emoji": ":family:",
```
Comment (ytjohn) [inline slackcat-example.conf]: Honestly, I never got the icon to change, whether I was using the config file, cli, or environment variable. So I'm unsure on the format here.

File: README.md [hunk 2]
```diff
@@ -45,6 +46,11 @@ slackcat will take each line from stdin and post it as a message to Slack:
 
     tail -F logfile | slackcat
 
+Be aware that if a file outputs blank lines, this will result in a 500 error from slack. You can remedy this using
+grep to filter out blank lines:
+
+    tail -F logfile | grep --line-buffered -v '^\s*$' | slackcat
```
Comment (codeincontext) [inline README.md:52]: Nice idea!

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+slackcat.conf
+.idea/
```
Comment (codeincontext) [inline .gitignore:2]: `.idea`?

**Comment by codeincontext** [comment]
Thanks for this!

I'm on a business trip and will look over it when I can. I'll want to tidy up the commits/merges as I think I already have some of them in my history. You can do that if you want, but I'm already grateful for your time spent on this 😀

File: .gitignore (hunk 3 above)
Comment (ytjohn) [inline .gitignore:2]: That's a directory created by IntelliJ IDEs (IntelliJ IDEA, PyCharm, etc).

Comment (ytjohn) [inline .gitignore:2]: Actually, if I had use their template, here is "common" stuff for .gitignore. 

Would you rather I: a) leave as is, b) remove the .idea line, or c) re-create with template below and comments?

```

### Go template
# Compiled Object files, Static and Dynamic libs (Shared Objects)
*.o
*.a
*.so

# Folders
_obj
_test

# Architecture specific extensions/prefixes
*.[568vq]
[568vq].out

*.cgo1.go
*.cgo2.c
_cgo_defun.c
_cgo_gotypes.go
_cgo_export.*

_testmain.go

*.exe
*.test
*.prof


### JetBrains template
# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion

*.iml

## Directory-based project format:
.idea/
# if you remove the above rule, at least ignore the following:

# User-specific stuff:
# .idea/workspace.xml
# .idea/tasks.xml
# .idea/dictionaries

# Sensitive or high-churn files:
# .idea/dataSources.ids
# .idea/dataSources.xml
# .idea/sqlDataSources.xml
# .idea/dynamic.xml
# .idea/uiDesigner.xml

# Gradle:
# .idea/gradle.xml
# .idea/libraries

# Mongo Explorer plugin:
# .idea/mongoSettings.xml

## File-based project format:
*.ipr
*.iws

## Plugin-specific files:

# IntelliJ
/out/

# mpeltonen/sbt-idea plugin
.idea_modules/

# JIRA plugin
atlassian-ide-plugin.xml

# Crashlytics plugin (for Android Studio and IntelliJ)
com_crashlytics_export_strings.xml
crashlytics.properties
crashlytics-build.properties


```

**Comment by ytjohn** [comment]
Good point on the commits. I had merged in the better-http-error before starting to work with your code changes. Since your code already has that merged in, maybe I should attempt to just pull your current master down and then merge in the proxy specific changes.

**Comment by codeincontext** [comment]
Yep, rebase for the win
- adam

**Comment by codeincontext** [comment]
Re. gitignore: Adding exceptions per editor seems like a can of worms. If you find this a problem generally, consider making a gitignore in your home directory
- adam

**Comment by ytjohn** [comment]
Ok, I created #2 to replace this one.

**Pull request closed without merging**